### PR TITLE
Pull request for Issue 140

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
I tried to have each commit do a similar thing.
I have not yet created the TFS variation that needs to happen in tandem.
All the tests were running for me.

### CODE REVIEW

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/src/Infrastructure.Common.csproj [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        </ProjectReference>
+    </ItemGroup>
+    <Import Project="testproperties.props" />
+    <Import Project="testproperties.targets" />
+    <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+    <!-- Add our code-gen target after the imports above because they reset $(CoreCompileDependsOn) -->
+    <PropertyGroup>
+        <CoreCompileDependsOn>$(CoreCompileDependsOn);CreateTestProperties</CoreCompileDependsOn>
+    </PropertyGroup>
+</Project>
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/src/Infrastructure.Common.csproj]: Nit: add newline at the end.  Not needed for this PR but next time you touch the file

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/src/project.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "dependencies": {
+    "System.Net.Sockets" : "4.0.10-beta-*",
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/src/project.json]: If this test still needs System.Net.Sockets, follow the example @mconnew provided in src\System.Private.ServiceModel\src\project.json.  It moves into its own special area at the end.

**Comment by roncain** [comment]
I buddy-built this and it passed all tests.  But to build successfully in VS I first had to delete all the packages.config files.

**Comment by roncain** [comment]
Before we merge, we will need to make a shelveset that fixes all the the wrapper projects in our experimental NET Native test framework.  We should try to merge and check that in at the same time.

**Comment by roncain** [comment]
This LGTM assuming these thing are done:
1. Verify deleting packages.config (except under Bridge) still passes all builds and can build in VS.
2. Put System.Net.Sockets in the right place in the project.json or remove it
3. Create/modify the wrapper projects we use to test against NET Native and verify this PR can run against those.

**Comment by StephenBonikowsky** [comment]
I'll get those three items done tomorrow @roncain

**Pull request merged**